charmcraft/linters.py

```diff
@@ -197,10 +195,48 @@ def run(self, basedir: pathlib.Path) -> str:
         return result
 
 
+class JujuMetadata:
+    """Check that the metadata.yaml file exists and is sane.
+
+    The charm is considered to have a valid metadata if the following checks are true:
+
+    - the metadata.yaml is present
+    - it is a valid YAML file
+    - it has at least the following fields: name, summary, and description
+    """
+
+    check_type = CheckType.lint
+    name = "metadata"
+    url = "https://juju.is/docs/sdk/charmcraft-analyze#heading--metadata"
+    text = "Problems found with metadata.yaml file."
+
+    # different result constants
+    Result = namedtuple("Result", "ok errors")(ok="ok", errors="errors")
+
+    def run(self, basedir: pathlib.Path) -> str:
+        """Run the proper verifications."""
+        try:
+            metadata = parse_metadata_yaml(basedir)
+        except Exception:
+            # file not found, corrupted, or mandatory "name" not present
```

Comment:
is checking the name here only a side effect of charmcraft needing it?

Reply:
Yes. We cannot build a charm without knowing its name.